### ISSUE TITLE
Undocumented changes when generating Kotlin code

### ISSUE THREAD

**Issue by pdeva** [jOOQ/jOOQ#11217]
### Expected behavior
I was generating java code uptil jooq 3.13. I recently upgraded to 3.14 and started using the Kotlin generator.
I noticed that it generates some new classes which are essential to using Jooq but not documented.

For example, this was my code (in kotlin) using Java generated Jooq (v 3.13):

```kotlin
 val r = dslContext
                .newRecord(Tables.AWS_INTEG.ACCOUNT_ID, Tables.AWS_INTEG.AWS_ACCOUNT_ID)
                .values(curUser.accountId, args.awsAccountId)

AwsIntegDao(jooqConfig).deleteById(r)
```

However, switching to the Kotlin generated code in jooq 3.14, the `Tables` class is gone entirely! I have to use a `Public` class, which is not shown in any of the examples.

This what my new code looks like:

```
val r = dslContext
                .newRecord(Public.PUBLIC.AWS_INTEG.ACCOUNT_ID,Public.PUBLIC.AWS_INTEG.AWS_ACCOUNT_ID)
                .values(curUser.accountId, args.awsAccountId)

        AwsIntegDao(jooqConfig).deleteById(r as Record2<UUID, String>)
```

Note that i also had to cast `r` to the non-nullable generic version of `Record2`, otherwise the type comes out to be `Record2<UUID?, String?>` which the `deleteById()` method wont accept. 

### Actual behavior

There should be documentation for all the changes in 3.14 and in switching to Kotlin code generation. there isnt one right now

### Versions

- jOOQ: 3.14.4
- Java: 11
- Database (include vendor): postgres
- OS: ubuntu 20
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your message.

A lot of decisions about the subtle or not so subtle changes when going from the `JavaGenerator` to the `KotlinGenerator` have been documented in this issue here while they were implemented:
https://github.com/jOOQ/jOOQ/issues/6248

Including:

> **Globally available objects should be package objects, not companion objects**

All the changes in 3.14 are documented in the release notes of 3.14.0 - 3.14.4 (so far): 
https://www.jooq.org/notes/?version=3.14

There, you'll also find all the fixed bugs so far. The bugs that we're not yet aware of (e.g. the `deleteById()` glitch, which is a separate issue: https://github.com/jOOQ/jOOQ/issues/11218), we obviously couldn't document yet. You refer to *all the changes in 3.14*, so I'm assuming you have discovered more issues? I'd love to hear about them! We'll address them as soon as possible.

The initial `KotlinGenerator` (prior to the release of 3.14.0) generated an `object Tables`, which corresponds to Java's `Tables` class with static members. Unfortunately, Kotlin doesn't support import on demand from objects, only from packages, see:
https://stackoverflow.com/q/33717490/521799

That makes the `object Tables` quite useless. This is why the decision has been made to generate a `com/example/db/tables/references/Tables.kt` file which contains package level constants for your tables

Yes, we need to document these differences to the JavaGenerator, this one being the most important one:
https://github.com/jOOQ/jOOQ/issues/6248#issuecomment-685557269

But as you can see here, the KotlinGenerator has been quite in motion over the past patch releases, and writing a tutorial might have been premature, because it would have been outdated quite quickly:
https://github.com/jOOQ/jOOQ/issues?q=is%3Aissue+label%3A%22C%3A+Integration%3A+Kotlin%22+is%3Aclosed

There will definitely be a tutorial. Thanks again for your message.

**Comment by lukaseder** [comment]
https://github.com/jOOQ/jOOQ/issues/11218 is fixed for 3.15.0 and 3.14.5
